sdk/src/main/scala/com/horizen/SidechainHistory.scala

```diff
@@ -130,21 +129,20 @@ class SidechainHistory private (val storage: SidechainHistoryStorage,
 
   def blockToBlockInfo(block: SidechainBlock): Option[SidechainBlockInfo] = storage.blockInfoOptionById(block.parentId).map(calculateBlockInfo(block, _))
 
-
   // Calculate SidechainBlock info based on passed block and parent info.
   private def calculateBlockInfo(block: SidechainBlock, parentBlockInfo: SidechainBlockInfo): SidechainBlockInfo = {
     val lastBlockInPreviousConsensusEpoch = getLastBlockInPreviousConsensusEpoch(block.timestamp, block.parentId)
     val nonceConsensusEpochInfo = getOrCalculateNonceConsensusEpochInfo(block.header.timestamp, block.header.parentId)
     val vrfOutputOpt = getVrfOutput(block.header, nonceConsensusEpochInfo)
-    val prevCumulativeCommTreeHash:FieldElement = parentBlockInfo.mainchainHeaderBaseInfo.last.cumulativeCommTreeHash
+    val prevBaseInfo:MainchainHeaderBaseInfo = storage.getLastMainchainHeaderBaseInfoInclusion(block.parentId)
```

Comment:
you may optimize this part: for the SC blocks with no MC headers -> don't search for `prevBaseInfo`.
For example: 
```
val blockMainchainHeaderBaseInfoSeq: Seq[...] = if(block.mainchainHeaders.isEmpty) Seq() else {
  val prevBaseInfo:MainchainHeaderBaseInfo = storage.getLastMainchainHeaderBaseInfoInclusion(block.parentId)
  MainchainHeaderBaseInfo.getMainchainHeaderBaseInfoFromBlock(block, prevBaseInfo.cumulativeCommTreeHash)
}
``` 
and pass `blockMainchainHeaderBaseInfoSeq` to the `SidechainBlockInfo` constructor.